sched/test_per_group_lb.py

```diff
@@ -0,0 +1,389 @@
+            'config' : NGINX_CONFIG,
+        },
+        {
+            'id' : 'nginx_8003',
+            'type' : 'nginx',
+            'port' : '8003',
+            'delay' : '0.015',
+            'status_uri' : 'http://${server_ip}:${port}/nginx_status',
+            'config' : NGINX_CONFIG,
+        },
+        {
+            'id' : 'nginx_8004',
+            'type' : 'nginx',
+            'port' : '8004',
+            'delay' : '0.02',
+            'status_uri' : 'http://${server_ip}:${port}/nginx_status',
+            'config' : NGINX_CONFIG,
+        },
+        {
+            'id' : 'nginx_8005',
+            'type' : 'nginx',
+            'port' : '8005',
+            'delay' : '0.025',
+            'status_uri' : 'http://${server_ip}:${port}/nginx_status',
+            'config' : NGINX_CONFIG,
+        },
+        {
+            'id' : 'nginx_8006',
+            'type' : 'nginx',
+            'port' : '8006',
+            'delay' : '0.03',
+            'status_uri' : 'http://${server_ip}:${port}/nginx_status',
+            'config' : NGINX_CONFIG,
+        },
+        {
+            'id' : 'nginx_8007',
+            'type' : 'nginx',
+            'port' : '8007',
+            'delay' : '0.035',
+            'status_uri' : 'http://${server_ip}:${port}/nginx_status',
+            'config' : NGINX_CONFIG,
+        },
+        {
+            'id' : 'nginx_8008',
+            'type' : 'nginx',
+            'port' : '8008',
+            'delay' : '0.04',
+            'status_uri' : 'http://${server_ip}:${port}/nginx_status',
+            'config' : NGINX_CONFIG,
+        },
+        {
+            'id' : 'nginx_8009',
+            'type' : 'nginx',
+            'port' : '8009',
+            'delay' : '0.1',
+            'status_uri' : 'http://${server_ip}:${port}/nginx_status',
+            'config' : NGINX_CONFIG,
+        },
+    ]
+
+    clients = [
+        {
+            'id' : 'wrk_vhost_1',
+            'type' : 'wrk',
+            'addr' : "${tempesta_ip}:80",
+        },
+        {
+            'id' : 'wrk_vhost_2',
+            'type' : 'wrk',
+            'addr' : "${tempesta_ip}:80",
+        },
+    ]
+
+    tempesta = {
+        'sched_global_opt' : "",
+        'sched_custom_opt' : "",
+        'sched_global_late_opt' : "",
+        'config' : TEMPESTA_CONFIG,
+    }
+
+    group_scheds = [('custom', 'static'), ('default', 'static')]
+
+    # Base precision for ratio static.
+    precision = 0.005
+    # Minimum request count delta used for short term tests.
+    min_delta = 10
+    # Min server weight for ratio dynamic scheduler.
+    min_server_weight = 30
+
+    def check_static_lb(self, servers, group_name):
+        """ Group of servers use ratio static load balancer. For details
+        see test_ratio_static.py .
+        """
+        exp_reqs = 0
+        delta = 0
+
+        for srv in servers:
+            tf_cfg.dbg(3,
+                       "Server %s received %d responses"
```

Comment:
Maybe `responses` -> `requests` (the same in assert message below).

Reply:
Fixed.